patches/joinrules/matrix-react-sdk+3.53.0.patch

```diff
@@ -0,0 +1,116 @@
+-                    });
+-                    return;
+-                }
+-
+-                setContent({
+-                    join_rule: JoinRule.Restricted,
+-                    allow: newAllowRoomIds.map(roomId => ({
+-                        "type": RestrictedAllowType.RoomMembership,
+-                        "room_id": roomId,
+-                    })),
+-                });
+-            };
+-
+-            const onEditRestrictedClick = async () => {
+-                const restrictedAllowRoomIds = await editRestrictedRoomIds();
+-                if (!Array.isArray(restrictedAllowRoomIds)) return;
+-                if (restrictedAllowRoomIds.length > 0) {
+-                    onRestrictedRoomIdsChange(restrictedAllowRoomIds);
+-                } else {
+-                    onChange(JoinRule.Invite);
+-                }
+-            };
+-
+-            description = <div>
+-                <span>
+-                    { _t("Anyone in a space can find and join. <a>Edit which spaces can access here.</a>", {}, {
+-                        a: sub => <AccessibleButton
+-                            disabled={disabled}
+-                            onClick={onEditRestrictedClick}
+-                            kind="link_inline"
+-                        >
+-                            { sub }
+-                        </AccessibleButton>,
+-                    }) }
+-                </span>
+-
+-                <div className="mx_JoinRuleSettings_spacesWithAccess">
+-                    <h4>{ _t("Spaces with access") }</h4>
+-                    { shownSpaces.map(room => {
+-                        return <span key={room.roomId}>
+-                            <RoomAvatar room={room} height={32} width={32} />
+-                            { room.name }
+-                        </span>;
+-                    }) }
+-                    { moreText && <span>{ moreText }</span> }
+-                </div>
+-            </div>;
+-        } else if (SpaceStore.instance.activeSpaceRoom) {
+-            description = _t("Anyone in <spaceName/> can find and join. You can select other spaces too.", {}, {
+-                spaceName: () => <b>{ SpaceStore.instance.activeSpaceRoom.name }</b>,
+-            });
+-        } else {
+-            description = _t("Anyone in a space can find and join. You can select multiple spaces.");
+-        }
+-
+-        definitions.splice(1, 0, {
+-            value: JoinRule.Restricted,
+-            label: <>
+-                { _t("Space members") }
+-                { upgradeRequiredPill }
+-            </>,
+-            description,
+-            // if there are 0 allowed spaces then render it as invite only instead
+-            checked: joinRule === JoinRule.Restricted && !!restrictedAllowRoomIds?.length,
+-        });
+-    }
++    /**
++     * :TCHAP: removed all code about "restricted" sharing, which has to do with spaces.
++     * See patches/matrix-react-sdk+<version>.patch for details.
```

Comment:
leave notes in the patched files so we know what it is about.

Reply:
Is there additional notes in the patch? I don't see them

Reply:
It's just that "TCHAP" comment. The patch is generated, so you can't really add comments directly in it. So I added the comments in the code file.